exporter/fileexporter/file_exporter.go

```diff
@@ -68,7 +70,9 @@ func exportMessageAsLine(e *fileExporter, message proto.Message) error {
 }
 
 func (e *fileExporter) Start(context.Context, component.Host) error {
-	return nil
+	var err error
+	e.file, err = os.OpenFile(e.path, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
```

Comment:
I think this is not equivalent to the previous code. This will open the file multiple times, once for each signal type. Not sure if it matters, maybe not.

Reply:
No, because of the SharedComponents. Only one instance will be created, and also it is guaranteed that file is created first time this is started.

There is a different issue that may happen (I don't know exactly if we want to fix it or not), right now Start of the "shared component" is called first time Start is called, and Shutdown the same, which means an exporter instance shared by Traces/Metrics/Logs will be closed when first of these signals is closed, but the way we shutdown the pipelines right now should not be a problem. We can fix this in the SharedComponent to do refcount.

Reply:
Also this PR fixes a problem, see the test changed in DefaultComponents to show the problem, Close was called multiple times so the shutdown of the entire collector will fail if a shared instance of this exporter is created.

Reply:
> There is a different issue that may happen (I don't know exactly if we want to fix it or not), right now Start of the "shared component" is called first time Start is called, and Shutdown the same, which means an exporter instance shared by Traces/Metrics/Logs will be closed when first of these signals is closed, but the way we shutdown the pipelines right now should not be a problem. We can fix this in the SharedComponent to do refcount.

I didn't think this through in details, but I think the new way is better. I don't know why one instance of exporter needs to start/stop more than once.